src/main/java/se/sundsvall/intricdatacollector/datasource/confluence/api/ConfluenceWebhookResources.java

```diff
@@ -45,13 +45,22 @@ class ConfluenceWebhookResources {
         produces = APPLICATION_JSON_VALUE
     )
     ResponseEntity<Void> handleWebhookEvent(@PathVariable("municipalityId") final String municipalityId, @RequestBody @Valid final ConfluenceWebhookData request) {
+        // Manually check if webhooks are enabled for the given municipality
+        boolean webhookEnabled = ofNullable(properties.environments().get(municipalityId))
+            .map(ConfluenceIntegrationProperties.Environment::webhook)
+            .map(ConfluenceIntegrationProperties.Environment.Webhook::enabled)
+            .orElse(false);
+        if (!webhookEnabled) {
+            throw Problem.valueOf(FORBIDDEN, "Webhooks are disabled for the requested municipality id");
+        }
+
         var eventType = EventType.fromString(request.eventType());
         var pageId = request.page().id().toString();
 
-        if (EnumSet.of(PAGE_CREATED, PAGE_UPDATED, PAGE_RESTORED).contains(eventType)) {
-            dataSource.processPage(municipalityId, eventType, pageId);
-        } else if (eventType == PAGE_REMOVED) {
-            dataSource.deletePage(municipalityId, pageId);
+        switch (eventType) {
+            case PAGE_CREATED, PAGE_RESTORED -> dataSource.insertPage(municipalityId, pageId);
+            case PAGE_UPDATED -> dataSource.updatePage(municipalityId, pageId);
+            case PAGE_REMOVED -> dataSource.deletePage(municipalityId, pageId);
         }
 
         return ok().build();
```

Comment:
Add `.header(CONTENT_TYPE, ALL_VALUE)` to prevent problem with WSO2 returning resource as octet stream otherwise (comment goes for all places where <Void> is returned?